workers/unity/Assets/Gdk/Core/Collections/ListProvider.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Collections.Generic;
+
+namespace Assets.Gdk.Core.Collections
+{
+    internal static class ListProvider<T>
```

Comment:
T isn't actually being used as a free parameter here, it's always in the form of List<T>. Similarly for the dictionary version. You could replace these both with a generic provider that just takes a single parameter.

Reply:
The only sticking point would be that you need to constrain it to `ICollection<U>` to call `Clear()`. Which can be done by adding another parameter, `U` as a sort of pseudo existential type. But because Dictionary is awkward in its definition of `T` you have to call it with `Provider<Dictionary<TKey, TValue>, KeyValuePair<TKey, TValue>.Get(handle)`.